internal/repository/prune.go

```diff
@@ -434,12 +444,15 @@ func decidePackAction(ctx context.Context, opts PruneOptions, repo *Repository,
 		}
 	}
 
-	if len(repackSmallCandidates) < 10 {
+	// calculate limit for number of unused bytes in the repo after repacking
+	maxUnusedSizeAfter := opts.MaxUnusedBytes(stats.Size.Used)
+	// wpl we need a mixed calculation based the existing condition and on --max-unused
+	if len(repackSmallCandidates) >= 10 || opts.RepackSmall && (uint64(sizeRepackSmallCandidates) >= maxUnusedSizeAfter) {
```

Comment:
Please revert this change. If users set `--max-unused 0`, your change causes repeated prune executions to repack the same data each time. The point of repacking small files is reduce the number of files, but a few files more or less won't matter. If `--max-unused` yields a large value (>= 1.28GB to be precise), then the new clause is irrelevant as `len(repackSmallCandidates) >= 10` will always be true first.